### PR TITLE
Tt 2466 refactor workflow

### PR DESCRIPTION
**Pull request by gtryus**

### CODE REVIEW

**Review by sarahentzel — CHANGES_REQUESTED** [review]

File: src/components/Workflow/ScriptureTable.tsx [hunk 1]
```diff
@@ -0,0 +1,830 @@
...
+  };
+
+  const doAssign = (where: number[]) => {
+    setAssignSections(where);
+    setAssignSectionVisible(true);
+  };
+  const handleAssign = (where: number[]) => () => {
+    if (changed || myChangedRef.current) {
+      startSave();
+      waitForSave(() => doAssign(where), 100);
```
[the remark this answers is not in the document]
Comment (gtryus) [reply src/components/Workflow/ScriptureTable.tsx:483]: I changed it to 500 and used a constant.

File: src/components/Workflow/ScriptureTable.tsx [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+        } else {
+          await localSave(workflow, sections, passages, prevSave);
+          return false;
+        }
+      };
+      if (numChanges > 50) setBusy(true);
+      let change = false;
+      let start = 0;
+      if (!offlineOnly) {
+        for (; start + 200 < numChanges; start += 200) {
```
[the remark this answers is not in the document]
Comment (gtryus) [reply src/components/Workflow/ScriptureTable.tsx:610]: yes, I changed to workflow.length

File: src/components/Workflow/ScriptureTable.tsx [hunk 3]
```diff
@@ -0,0 +1,830 @@
...
+      setWorkflow([...workflow]);
+      setChanged(true);
+    }
+  };
+
+  const handleTranscribe = (i: number) => {
+    const { wf } = getByIndex(workflow, i);
+    const id = wf?.passageId?.id || '';
+    const passageRemoteId = remoteIdNum('passage', id, memory.keyMap) || id;
+    if (changed || myChangedRef.current) {
```
[the remark this answers is not in the document]
Comment (gtryus) [reply src/components/Workflow/ScriptureTable.tsx:458]: changed to just use myChangedRef.current

File: src/components/Workflow/PlanSheet.tsx [hunk 4]
```diff
@@ -296,37 +283,14 @@ export function PlanSheet(props: IProps) {
...
-    );
-  };
-
-  const numCol = [3, 5]; // Section num = col 2, Passage num = col 4
-  const handleCellsChanged = (changes: Array<IChange>) => {
+  const handleCellsChanged = (changes: Array<ICellChange>) => {
     if (readonly) return; //readonly
-    const grid = data.map((row: Array<ICell>) => [...row]);
-    changes.forEach(({ cell, row, col, value }: IChange) => {
-      if (row !== 0 && numCol.includes(col) && value && !isNum(value)) {
```
[the remark this answers is not in the document]
Comment (gtryus) [reply src/components/Workflow/PlanSheet.tsx:316]: Handled this in updateData. If invalid, it puts up a message and resets it to the previous value.

**Pull request merged**